lib/private/Template/JSResourceLocator.php

```
@@ -75,9 +75,14 @@ public function doFind($script) {
 		$app_path = \OC_App::getAppPath($app);
 		$app_url = \OC_App::getAppWebPath($app);
 
+		// Account for the possibility of having symlinks in app path. Doing
+		// this in a separate variable, because an empty argument to realpath
+		// gets turned into cwd, which makes it hard to see if app_path got set.
+		$real_app_path = realpath($app_path);
```

Shouldn't we also do this after the check if the app_path is empty like it is done in in the CSSResourceLocator? Otherwise $real_app_path still might be the working directory for empty app paths.

---

Yeah the diff is hiding that from you, expand is a little below this. It's done [here](https://github.com/nextcloud/server/pull/7061/files#diff-e75d752a35a50ba300698850b35f80efR89).

---

Although this will cause weird behavior with using the cwd for l10n, huh. I refactored it just a tad to account for that.

---

Can't you just move the check got app_path and app_url in front of the realpath call? That way we don't append any js sources at cwd.

---

Not without changing the `l10n` behavior. I don't claim to be an expert in the code, of course, but right now if a `l10n` file doesn't exist, the error is ignored. If we move the check above that logic, we start raising an error instead.

---

Ah, ok, makes sense to keep it like this i guess. 👍